gnocchi/indexer/sqlalchemy.py

```diff
@@ -568,25 +568,33 @@ def get_archive_policy(self, name):
         with self.facade.independent_reader() as session:
             return session.query(ArchivePolicy).get(name)
 
-    def update_archive_policy(self, name, ap_items):
-        with self.facade.independent_writer() as session:
-            ap = session.query(ArchivePolicy).get(name)
-            if not ap:
-                raise indexer.NoSuchArchivePolicy(name)
-            current = sorted(ap.definition,
-                             key=operator.attrgetter('granularity'))
-            new = sorted(ap_items, key=operator.attrgetter('granularity'))
-            if len(current) != len(new):
+    def update_archive_policy(self, name, ap_items, new_name=None):
+        ap = self.get_archive_policy(name)
+        if not ap:
+            raise indexer.NoSuchArchivePolicy(name)
```

Comment:
I don't know if it has a great impact, but changing this creates 2 session rather than one.